operator-common/src/main/java/io/strimzi/operator/common/operator/resource/ConfigParameter.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common.operator.resource;
+
+import io.strimzi.operator.common.InvalidConfigurationException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Models a configuration parameter, identified by a unique key, which may be required, and if not may have a default value.
+ * Optional parameters without a default value implicitly have a null default.
+ * The key is also the name of the environment variable from which the value may be read, when it is read from the environment.
+ *
+ * @param key Configuration parameter name/key
+ * @param <T> Type of object
+ * @param type type of the default value
+ * @param defaultValue default value of the configuration parameter
+ * @param required  If the value is required or not
+ */
+public record ConfigParameter<T>(String key, AbstractConfig<T> type, String defaultValue, boolean required, Map<String, ConfigParameter<?>> map) {
+    /**
+     * Contructor
+     * @param key Configuration parameter name/key
+     * @param type type of the default value
+     * @param defaultValue default value of the configuration parameter
+     * @param required  If the value is required or not
+     * @param map Configuration map
+     */
+    public ConfigParameter {
+        map.put(key, this);
+    }
+
+    /**
+     * Generates the configuration map
+     * @param envVarMap    Map containing values entered by user.
+     * @param configValues Map containing all the configuration keys with default values
+     * @return             Generated configuration map
+     */
+    public static Map<String, Object> define(Map<String, String> envVarMap, Map<String, ConfigParameter<?>> configValues) {
```

Comment:
`configValues` should probably be `configParameters`, since it doesn't contain any actual _values_, but rather the schema for a set of expected values.